linters/import/main.go

```diff
@@ -0,0 +1,438 @@
+
+	pathLit := is.Path
+	if pathLit != nil {
+		path = pathLit.Value
+	}
+
+	return importSpec{
+		Name: filename,
+		Path: path,
+		Line: line,
+	}
+}
+
+func findErrors(importDecls []importDecl, patterns []string, file *ast.File) lintErrors {
+	var (
+		lintErrors lintErrors
+		// checkedPatterns is used to keep track of patterns we've seen, but want to use again
+		checkedPatterns []string
+
+		// seenPatterns is used to keep track of patterns that have been seen already
+		seenPatterns = make(map[string]interface{})
+	)
+
+	// check if there is more than one import group
+	if len(importDecls) > 1 {
+		lintErr := lintError{
+			fileName: importDecls[0].Groups[0].Imports[0].Name,
+			err:      errMultipleImport,
+		}
+		lintErrors = append(lintErrors, lintErr)
+		return lintErrors
+	}
+
+	return findRecErr(importDecls[0].Groups, checkedPatterns, patterns, lintErrors, seenPatterns)
+}
+
+func findRecErr(importGroup []importGroup, checkedPatterns, currentPatterns []string, lintErrors lintErrors, seenPatterns map[string]interface{}) lintErrors {
+	for _, ind := range importGroup {
+		checkedPatterns = checkedPatterns[:0]
+		var checking bool
+		for idx, i := range ind.Imports {
+			if len(currentPatterns) == 0 && len(importGroup) > 0 && len(checkedPatterns) > 0 {
+				lintErrors = addLintError(i, lintErrors, errNoMatch, "")
+
+				empty := checkedPatterns[:0] // clear out checkedPatterns before moving on to groups (could probably get rid of this)
+				return findRecErr(importGroup[1:], empty, checkedPatterns, lintErrors, seenPatterns)
+			}
+			if len(currentPatterns) == 0 && len(importGroup) > 0 {
+				return addLintError(i, lintErrors, errTooManyImports, "")
+			}
+			if currentPatterns[0] == standard {
+				match, err := regexp.MatchString(`\.`, i.Path)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Unable to match: pattern, got error %v", err)
+					return nil
+				}
+				if !match {
+					if _, ok := seenPatterns[currentPatterns[0]]; ok {
+						lintErrors = addLintError(i, lintErrors, errGroupMachedAlready, currentPatterns[0])
+						return findRecErr(importGroup[1:], checkedPatterns, currentPatterns[1:], lintErrors, seenPatterns)
+					}
+					checking = true
+				}
+				if match && checking {
+					lintErrors = addLintError(i, lintErrors, errOutOfOrder, currentPatterns[0])
+				}
+				if match && !checking {
+					return findRecErr(importGroup[0:], checkedPatterns, currentPatterns[1:], lintErrors, seenPatterns)
+				}
+				if idx == len(ind.Imports)-1 {
+					seenPatterns[currentPatterns[0]] = struct{}{}
+					return findRecErr(importGroup[1:], checkedPatterns, currentPatterns[1:], lintErrors, seenPatterns)
+				}
+			}
+			if currentPatterns[0] == external {
+				match, err := regexp.MatchString(`\.`, i.Path)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Unable to match: pattern, got error %v", err)
+					return nil
+				}
+				if match {
+					for seen := range seenPatterns {
+						match, err := regexp.MatchString(seen, i.Path)
```

Comment:
probably better to just do [strings.HasPrefix](https://golang.org/pkg/strings/#HasPrefix) instead of `regexp.MatchString` here

Reply:
I don't think this will work because the imports actually start with `"` so we'd either need to add a `"` to the `prefix` that it checks against or remove the `"` from the import.